tests/test_api_requests.py

```diff
@@ -213,6 +215,67 @@ async def test_job_id(self, job_id):
             assert "zyte-api" in resp.flags
             assert resp.body == f"<html>{job_id}</html>".encode("utf8")
 
+    @pytest.mark.parametrize(
+        "method",
+        [
+            "DELETE",
+            "GET",
+            "OPTIONS",
+            "PATCH",
+            "POST",
+            "PUT",
+            "TRACE",
+        ],
+    )
+    async def test_supported_method(self, method):
```

Comment:
What do you think about adding the following? as well as only importing `AyncMock` when its `Python3.8+`?
 
```python
@pytest.mark.skipif(
    sys.version_info < (3, 8), reason="Python3.7 has poor support for AsyncMocks"
)
```

In this way, we could pass the tests.

An alternative would be to have workarounds for `AyncMock` like [this one](https://github.com/scrapinghub/web-poet/blob/f2dce80c7c7fc0f3652270843bf01c9b957c4709/tests/test_requests.py#L15-L24)  and [this one](https://github.com/scrapinghub/scrapy-poet/pull/62/files#diff-9e08e8c7769db7b58089fd7659d75ca96df40bb7b5d50299d3a30abd54da9ad6R90-R94).